changelogs/unreleased/elections-pre-vote.md

```diff
@@ -8,3 +8,8 @@
 
  * Added `leader_idle` field to `box.info.election` table. This value shows time
    in seconds since the last communication with a known leader (gh-6654).
+
+## bugfix/raft
+
+ * Fixed `box.ctl.promote()` entering an infinite election loop when a node
```

Comment:
But why would it start the infinite loop? Does it mean that if during an election we lost the healthy quorum, the pre-vote check stops working? Maybe interrupt the election when the healthy quorum is lost? At least do not start a new round. Then a bigger issue would be fixed than with only manual promotion.

Reply:
There's no issue with automatic elections: if healthy quorum is lost, the node won't start another term (it won't stop being candidate in current term as well, but I think that's not an issue: the term is already bumped anyway).

Manual elections are fixed separately, because they start the candidate no matter what via `raft_promote()` and make it run for a single term. Then the cycle repeats.
I've made it so the elections are started once, even when there's no quorum, and then no re-elections are triggered if quorum is lost. I thinks it's ok that manual promote ignores pre-vote checks. A user should see some result of his actions (at least a term bump) after promotion.

Reply:
Thanks for the explanation, clear now.